src/librustc_errors/emitter.rs

```diff
@@ -458,7 +458,11 @@ impl ColorConfig {
                 }
             }
             ColorConfig::Never => ColorChoice::Never,
-            ColorConfig::Auto if atty::is(atty::Stream::Stderr) => {
+            ColorConfig::Auto
+                if env::var("CLICOLOR_FORCE").unwrap_or("0".to_string()) != "0"
```

Comment:
This unnecessarily allocates a string. Something like `env::var(...).as_ref().map(|s| &**s).unwrap_or("0")` should work. Also I think it would be nicer to use `ColorConfig::Auto => { if { ... } else { ... } }` instead.